### ISSUE TITLE
to_toml() ValueError: Circular reference detected

### ISSUE THREAD

**Issue by next-franciscoalgaba** [fabiocaccamo/python-benedict#53]
**Python version**

3.7.0

**Package version**

0.23.2

**Current behavior (bug description)**

After reading and modifying a TOML value in a benedict object. If trying to write the changes back to the TOML a ValueError is raised.

Example:
``` python
from benedict import benedict
pyproject_toml = benedict("./pyproject.toml", format="toml")
pyproject_toml["tool.poetry.name"] = "name"
pyproject_toml.to_toml()
```

Execption raised:

``` bash
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../.venv/lib/python3.7/site-packages/benedict/dicts/io/io_dict.py", line 223, in to_toml
    return self._encode(self, 'toml', **kwargs)
  File "/.../.venv/lib/python3.7/site-packages/benedict/dicts/io/io_dict.py", line 55, in _encode
    s = io_util.encode(d, format, **kwargs)
  File "/.../.venv/lib/python3.7/site-packages/benedict/dicts/io/io_util.py", line 28, in encode
    s = serializer.encode(d, **kwargs)
  File "/.../.venv/lib/python3.7/site-packages/benedict/serializers/toml.py", line 20, in encode
    data = toml.dumps(d, **kwargs)
  File "/.../.venv/lib/python3.7/site-packages/toml/encoder.py", line 67, in dumps
    raise ValueError("Circular reference detected")
ValueError: Circular reference detected
```

**Expected behavior**

New changed TOML dumped to stdout as string and if `filepath` argument specified, write to destination filepath.

The bug is identified in the code and a PR to fix the problem is going to be submitted ASAP.

**Comment by fabiocaccamo** [comment]
@next-franciscoalgaba thank you for reporting this, could you please provide your toml file?

**Comment by next-franciscoalgaba** [comment]
Hi, just created the PR fixing the issue: https://github.com/fabiocaccamo/python-benedict/pull/54

**Comment by next-franciscoalgaba** [comment]
> @next-franciscoalgaba thank you for reporting this, could you please provide your toml file?

The TOML file is a standard Poetry pyproject.toml file.

``` toml
[build-system]
requires = [ "poetry-core>=1.0.0",]
build-backend = "poetry.core.masonry.api"

[tool.poetry]
name = "hermes"
version = "0.5.21"
description = "CI tool using Poetry"
authors = [ "Francisco Algaba <...>",]
readme = "README.md"

[tool.poetry.scripts]
hermes = "hermes.main:app"

[tool.poetry.dependencies]
python = ">=3.6.1,<4.0.0"
python-benedict = "^0.23.2"
mkdocs-material = "^7.1.2"
mkdocstrings = "^0.15.0"
mkdocs = "^1.1.2"
shiv = "^0.4.0"
pytest-json-report = "^1.2.4"
cruft = "^2.8.0"

[tool.poetry.dev-dependencies]
pytest = "^5.2"
Pygments = "^2.8.1"
pre-commit = "^2.12.0"
isort = "^5.8.0"

[tool.poetry.dependencies.typer]
extras = [ "all",]
version = "^0.3.2"

[tool.poetry.dev-dependencies.black]
version = "^20.8b1"
allow-prereleases = true

```

Thank you for the fast reply.

**Comment by fabiocaccamo** [comment]
@next-franciscoalgaba I think that the problem is related to the `keypath_separator`, could you try to do this:
```python
from benedict import benedict
pyproject_toml = benedict("./pyproject.toml", format="toml", keypath_separator=None)
pyproject_toml["tool.poetry.name"] = "name"
pyproject_toml.to_toml()
```

**Comment by next-franciscoalgaba** [comment]
> @next-franciscoalgaba I think that the problem is related to the `keypath_separator`, could you try to do this:
> 
> ```python
> from benedict import benedict
> pyproject_toml = benedict("./pyproject.toml", format="toml", keypath_separator=None)
> pyproject_toml["tool.poetry.name"] = "name"
> pyproject_toml.to_toml()
> ```
> 
> More info [here](https://github.com/fabiocaccamo/python-benedict#keypath).

Exactly, the problem is related to the `keypath_separator`. Regarding my problem, that is parsing a Poetry project file, the standard for the framework for `keypath_separator` is `.`

With the submitted PR, the `keypath_separator` error is solved. Just submitted the required tests.

Thanks.

**Comment by fabiocaccamo** [comment]
@next-franciscoalgaba this has been fixed in `0.24.0` version.
